service/src/main/kotlin/fi/espoo/evaka/placement/PlacementController.kt

```diff
@@ -221,6 +225,54 @@ class PlacementController(
                         )
                         .also {
                             generateAbsencesFromIrregularDailyServiceTimes(tx, now, body.childId)
+
+                            // Only proceed with future absence and reservation deletion if
+                            // placements range is in future
+                            if (body.endDate.isAfter(now.toLocalDate())) {
+                                val range =
+                                    DateRange(
+                                        when (body.startDate.isAfter(now.toLocalDate())) {
+                                            true -> body.startDate
+                                            false -> now.toLocalDate().plusDays(1)
+                                        },
+                                        body.endDate
+                                    )
```

Comment:
This could just be 
```
val future = DateRange.of(now.toLocalDate().plusDays(1), null)
val range = FiniteDateRange(body.startDate, body.endDate).intersection(future)
```

`.intersection()` It returns `null` if the ranges don't overlap, i.e. the range would be in the past.

Reply:
The code could be even by changing this:
```kotlin
if (body.endDate.isAfter(now.toLocalDate())) {
    val future = DateRange(now.toLocalDate().plusDays(1), null)
    val range =
        DateRange(body.startDate, body.endDate).intersection(future)
    ...
```
to this:
```kotlin
val future = DateRange(now.toLocalDate().plusDays(1), null)
val range = DateRange(body.startDate, body.endDate).intersection(future)
if (range != null) {
    ...
```
`.intersection()` returns `null` if the ranges don't intersect, that is, if `body.endDate` is today or in the past.